---
  title: "Análisis bioinformático de la diversidad de zooplancton en el Golfo de México usando el marcador COI [Leray primers]"
author: "Grupo Bioinformática"
date: "Informe `r format(Sys.Date())`"
output: word_document

---
  
```{r setup, include=FALSE}
# Setting global option in all the chunks-code
knitr::opts_chunk$set(warning=FALSE, echo = FALSE, message = FALSE, dpi=300,
                      fig.width=12, fig.height=8)

rm(list=ls())

```

## Generalidades del análisis

```{r paths and colors}
set <- ''
path <- "/Users/cigom/metagenomics/COI/MULTIRUN/multirun_20190829_AMB_XIXIM/"


track.dir <- path
fasta.dir <- track.dir


scale2 <- c("Domain"="#edf8b1",  "Kingdom"="#7fcdbb", "Phylum"="#2c7fb8",  
            "Class"="#feb24c",  "Order"="#addd8e",  "Family"="#31a354",
            "Genus"="#bcbddc", "Species"="#756bb1")

Ranks <- c("Root","Dominio","Reino", 
           "Filo", "Clase", "Orden", 
           "Familia", "Genero", "Especie")

set <- "AMB"
```

```{r inputs name}
ctb.file = "multirun_ASVs_count.table"
tax.file = "multirun_ASVs.ALL.wang.taxonomy"
fasta.file = "multirun_ASVs.fasta"
multirunRD = 'multiresults.RData'


url <- 'https://raw.githubusercontent.com/RJEGR/metagenomics/master/readtx.R'
source(url)
```

```{r read.obj, echo = FALSE}
# Path files for processing within phyloseq
ctb.obj <- list.files(path, pattern = ctb.file, full.names = TRUE)

seqtab.nochim <- read.table(ctb.obj, row.names = 1)

names(seqtab.nochim) <- gsub("[.]", "-", names(seqtab.nochim), perl = TRUE)

# Taxonomy is processed this time with cigom_coi_detected2worms.R script

tax.path <- list.files(path, pattern = tax.file, full.names = TRUE)
tax <- read_rdp(tax.path)
colnames(tax) <- c(Ranks, "SL")

tax[is.na(tax)] <- 'Undetermined_R'

```

```{r insilico-trimming, echo = FALSE}
newenv <- new.env()
rdata.file <- list.files(path, pattern = multirunRD, full.names = TRUE)
load(file=rdata.file, env=newenv)

seqtab <- newenv$seqtab
# seqtab.nochim <- newenv$seqtab.nochim

require(dada2)
require(ggplot2)

len_df <- data.frame(nchar(getSequences(seqtab)))
names(len_df) <- c("Length")
len_df$Process <- "Nochim"


targetLength <- seq(300,320)
seqtab2.nochim <- seqtab.nochim[,nchar(colnames(seqtab.nochim)) %in% targetLength]


table(nchar(getSequences(seqtab2.nochim)))

# New distribution of Lengths
newlen_df <- data.frame(nchar(getSequences(seqtab2.nochim)))
names(newlen_df) <- c("Length")
newlen_df$Process <- "trimmed"

lens_df <- rbind(len_df, newlen_df)

# plot

lens_plot <- ggplot(lens_df, aes(Length, color=Process)) + 
  geom_freqpoly(binwidth=1, size=1, alpha=0.7) +
  scale_color_manual(values = c("#999999", "#E69F00"))+ 
  labs(title=paste0(". ESVs length distribution")) +
  theme_minimal() +
  theme(legend.position = c(0.05,0.95),
        legend.justification = c(0,1)) + geom_vline(xintercept = c(min(targetLength),max(targetLength)), linetype="dotted") #xlim(c(300,320))

lens_plot

png(paste0(path,"/", "multirun_lendistrib.png"), units="px", width=2900, height=2600, res=400)
plot(lens_plot, col=adjustcolor("black", alpha=0.2))
dev.off()


```

```{r metadata, echo = FALSE}
rdata.file <- list.files(path, pattern = multirunRD, full.names = TRUE)

newenv <- new.env()
load(file=rdata.file, env=newenv)
track_m <- newenv$track_m

samples.out <- subset(track_m, Process == 'merged')$Sample
run <- subset(track_m, Process == 'merged')$Factor
cruice <- sapply(strsplit(samples.out, "-"), `[`, 1)
subject <- sapply(strsplit(samples.out, "-"), `[`, 2)
marker <- 'COI'

transect <- substr(subject,1,1)
sample_type <- sapply(strsplit(samples.out, "\\-"), `[`, 3) 

samples <- data.frame(Corrida = run, 
                        Crucero=cruice, 
                        Estación=subject, 
                        Transecto=transect, 
                        muestra=sample_type, 
                        marcador =  marker)

rownames(samples) <- samples.out
```

```{r phyloseq.obj, echo = FALSE}
# # #
# We now construct a phyloseq object directly from the dada2 outputs.
# # #
library(phyloseq)

identical(names(seqtab.nochim),rownames(samples))
identical(rownames(seqtab.nochim), rownames(tax))

phyloseq = phyloseq(otu_table(seqtab.nochim, taxa_are_rows = TRUE), 
                    tax_table(as(tax, 'matrix')), 
                    sample_data(samples))

```

```{r clean.obj}
# preprocessing count-matrix

# head(as(otu_table(phyloseq), "matrix")[,26]) # compare it ...
# head(as(otu_table(phyloseq), "matrix")[,26] > 2)  # with this .. then:
# length(apply(as(otu_table(phyloseq), "matrix") > 2, MARGIN = 1, sum)) # size of non-singletones

# physeq <- subset_samples(phyloseq, Crucero == set) # Subset_samples by cruice
physeq <- prune_taxa(taxa_sums(phyloseq) > 1, phyloseq) # # Removing singletones taxa of abundance of zero

keepTaxa = apply(X = as(otu_table(physeq), "matrix") > 0, # Removing any abundance of zero
                 MARGIN = 1, FUN = sum) > 2 # Remove ASVs not k greater than k (2L) 

# table(keepTaxa)

clean.obj = prune_taxa(keepTaxa, physeq)
```

```{r write.table, eval = FALSE, echo = FALSE}
save <- psmelt(clean.obj)
# save <- psmelt(physeq)
colnames(save)[1] <- "ASV"
write.csv(save, file = paste0(path, "/", set, "taxones_identificados.csv"), 
          row.names = FALSE, quote = FALSE)
rm(save)
```

## Indice de figuras

```{r transform.sample.counts}
# Transform dataset to relative abundance
physeq2 <- NULL
# usar clean.obj o quitar zeros de physeq:
#physeq2 <- clean.obj
physeq2 <- transform_sample_counts(physeq, function(x) (x / sum (x) ) )
# physeq2 <- phyloseq::filter_taxa(physeq2, function(x) var(x) > 1e-5, TRUE)
physeq2 <- transform_sample_counts(physeq2, function(x) x * 100 )
physeq2

```

## Clasificación taxonómica y abundancia relativa

```{r Figura 6. Heatmap.phy, fig.align = "center", fig.cap = "Figura 6. Composición de las comunidades a nivel Filo a lo largo de las estaciones. Se presentan los taxones con varianza mayor a  1e-5.", out.width='\\textwidth', fig.width=8}
# Agglomerate taxa of the same Specie type [phylotyping] // demora 1-2 min //

Phylum <-tax_glom(physeq2, taxrank="Filo")
Phylum = subset_taxa(Phylum, Filo!="Undetermined_R")

library(ggplot2)

# extra figures
colourCount = length(unique(tax_table(Phylum)[,'Filo']))
#colourCount = nsamples(plotbar)
library(RColorBrewer)

getPalette = colorRampPalette(brewer.pal(colourCount, "Paired"))

theme_set( theme_classic())

mdf = psmelt(Phylum)

p = ggplot(mdf, aes(x = Estación, y = Abundance, fill = Filo)) +
  scale_fill_manual(values = getPalette(colourCount)) + coord_flip()
p = p + geom_bar(stat = "identity", position = "stack", color = "black")
p = p + theme(axis.text.x = element_text(angle = -90, hjust = 0), 
              axis.text.y = element_text(size  = 4.5))

bar <- p + facet_grid(Crucero ~ ., space = "free", scales = "free") +
  labs(x = 'Estación', y = 'Abundancia Relativa (%)', 
         caption = "Abundancia relativa de taxones;  Se remueven las asignaciones 'Undetermined' por muestra") +
  guides(fill=guide_legend(ncol=1))

png(paste0(path,"/", "Figura-6S-bar.phy.png"), units="px", width=3600, height=3200, res=400)
plot(bar, col=adjustcolor("black", alpha=0.2))
dev.off()

```

```{r Figura 7. barplot.class, fig.align = "center", fig.cap = "Figura 7. Composición de las comunidades a nivel Clase a lo largo de las estaciones. Se presentan los taxones con varianza mayor a  1e-5.", out.width='\\textwidth', fig.width=8}

Class <-tax_glom(physeq2, taxrank="Clase")
Class = subset_taxa(Class, Clase!="Undetermined_R")

colourCount = length(unique(tax_table(Class)[,'Clase']))

library(RColorBrewer)

getPalette = colorRampPalette(brewer.pal(colourCount, "Paired"))

mdf = psmelt(Class)

p = ggplot(mdf, aes(x = Estación, y = Abundance, fill = Clase)) +
  scale_fill_manual(values = getPalette(colourCount)) + coord_flip()
p = p + geom_bar(stat = "identity", position = "stack", color = "black")
p = p + theme(axis.text.x = element_text(angle = -90, hjust = 0), 
              axis.text.y = element_text(size  = 4))

bar <- p + facet_grid(Crucero ~ ., space = "free", scales = "free") +
  labs(x = 'Estación', y = 'Abundancia Relativa (%)', 
         caption = "Abundancia relativa de taxones;  Se remueven las asignaciones 'Undetermined' por muestra") +
  guides(fill=guide_legend(ncol=2))



png(paste0(path,"/", "Figura-7-bar.Class.png"), units="px", width=3000, height=3100, res=400)
plot(bar, col=adjustcolor("black", alpha=0.2))
dev.off()
```

```{r Figura 8. Heatmap.Or, fig.align = "center", fig.cap = "Figura 8. Composición de las comunidades a nivel Orden a lo largo de las estaciones. Se presentan los taxones con abundancia relativa mayor 1 %.", fig.height = 7, fig.width = 8}

Order <- tax_glom(physeq2, taxrank="Orden")

n_undet <- 1 - ntaxa(subset_taxa(Order, Orden!="Undetermined_R")) / ntaxa(Order)

Order <- subset_taxa(Order, Orden!="Undetermined_R")

Order_ <- phyloseq::filter_taxa(Order, function(x) sum(x) >= 1, TRUE)
taxaOrder = names(sort(taxa_sums(Order_), decreasing = TRUE))
otu_table(Order_)[otu_table(Order_) == 0] <- NA

#taxaOrder = rownames(sort(unique(tax_table(Order)[,'Orden']), decreasing=TRUE))

plot <- NULL

plot <- plot_heatmap(Order_, 
             sample.label = "Estación", 
             sample.order = "Transecto",
             taxa.label = "Orden",
             taxa.order = taxaOrder,
             na.value = "#e6e7ed", trans = NULL, low = "white", high = "#081D58") +
        labs(fill = "Abundancia\nRelativa (%)", title = paste0(set, " Crucero"),
             caption = "Se presentan taxones con abundancia relativa >= 1 % ") +
        theme(legend.position = "left")

plot <- plot + coord_flip() + theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1), axis.text.y = element_text(size  = 4)) + facet_grid(Crucero ~ ., space = "free", scales = "free")

png(paste0(path,"/", "Figura-8-Heatmap2.Or.png"), units="px", width=4000, height=2400, res=400)
plot(plot, col=adjustcolor("black", alpha=0.2))
dev.off()
```

```{r Figura 10. Boxplot species, fig.align = "center", fig.cap = "Especies más abundantes a lo largo de todas las estaciones. Se muestran aquellos taxones que representan al menos 10% de la abundancia total por muestra.", fig.width = 8, fig.height = 7}
library(data.table)
library(ggplot2)
# also try: https://github.com/joey711/phyloseq/issues/494

Level <- length(Ranks)
Rank <- Ranks[Level]

glom <- physeq2
glom <- tax_glom(glom, taxrank = Rank)
# create dataframe from phyloseq object
dat <- data.table(psmelt(glom))

dat.bkp <- dat
dat$Especie <- as.character(dat$Especie)

size <- 1
# Change name to remainder of rank less than 1%
dat[(Abundance <= size), Especie := "Otros"]

# boxplot
plot <- NULL
max_ab <- round(max(dat$Abundance))

plot <- ggplot(dat[Abundance > 0 & Especie != 'Undetermined_R'],
      aes(x=reorder(Especie, Abundance, FUN = median),
          y=Abundance)) +
 geom_boxplot() +
 coord_flip() +
 #scale_y_log10() + 
 ylim(0, max_ab) +
 theme_classic() + 
 theme(axis.text.y = element_text(size  = 4)) +
 facet_grid(~ Crucero, space = "free", scales = "free") +
 labs(y = "Abundancia Relativa (%)", x ="Especies", title = paste0("Especies abundantes del crucero ", set),
      caption = paste0("taxones que representan mas del ", size, "% de la abundancia por muestra"))


png(paste0(path,"/", "Figura-10-boxplot.Sp.png"), units="px", width=2700, height=3700, res=400)
plot(plot, col=adjustcolor("black", alpha=0.2))
dev.off()
```

```{r Figura 10S. Bubbleplot., fig.align = "center", fig.cap = "Generos más abundantes a lo largo de todas las estaciones. Se muestran aquellos taxones presentes en almenos ??? de la abundancia total por muestra (Se colorean por orden cada una de las especies).", fig.width = 8, fig.height = 7}

dat0 <- dat.bkp # Este es el objeto aglomerado de especie

dat0$Especie <- as.character(dat0$Especie)

size <- 1
# Change name to remainder of rank less than 1%
dat0[(Abundance <= size), Especie := "Otros"]

dim(dat <- data.frame(dat0[Abundance > 0 & Especie != 'Otros' & Especie != 'Undetermined_R']))

sort_lev <- c(unique(dat[order(dat$Orden), 'Especie']))
dat$Especie <- factor(dat$Especie, levels = rev(sort_lev))

library(RColorBrewer)
getPalette = colorRampPalette(brewer.pal(12, "Paired"))

colourCount <- length((unique(dat$Clase)))

bbplot <- ggplot(dat, 
       aes(x = Estación, y = Especie, color = Clase)) +
  geom_point(aes(size=Abundance)) + 
  facet_grid(~Crucero, space = "free", scales = "free") +
  scale_color_manual(values = getPalette(colourCount)) +
  #scale_color_brewer(palette = 'Paired') +
  theme_classic() + guides(fill=guide_legend(ncol=1)) +
  theme(
        axis.title.x=element_blank(),
        axis.text.x=element_text(angle = 90, hjust = 1, size = 5),
        axis.text.y = element_text(size  = 5))

png(paste0(path,"/", "Figura-10-bubbleplot2.Sp.png"), units="px", 
    width=4500, height=3700, res=400)
plot(bbplot, col=adjustcolor("black", alpha=0.2))
dev.off()
```


```{r Figura 11 Rarefaccion, echo = FALSE, fig.align = "center", fig.cap = "Figura 11. Curva de rarefacción por muestra. Se presenta el número de amplicones a lo largo de las estaciones en relación al número de secuencias.",  out.width='\\textwidth', fig.width = 7, fig.height = 4}
# https://github.com/gauravsk/ranacapa/
# and https://github.com/joey711/phyloseq/issues/143

library(ranacapa)
theme_set(theme_classic())
getPalette = colorRampPalette(brewer.pal(3, "Paired"))

rr <- phyloseq
minsample <- min(sample_sums(rr))

rrplot <- ggrare(rr, step = 100, label = "Estación", 
                 color="Crucero", 
                 se = FALSE, plot = FALSE) 

plot <- NULL
plot <- rrplot +
  xlab("Numero de secuencias") + ylab("Numero de amplicones") + 
    theme(text = element_text(size=12),
        panel.background = element_blank(), legend.position = "top") +
  geom_point(size=0.7) +
  #geom_vline(xintercept = min(sample_sums(rr)), linetype = "dashed") +
  scale_color_manual(values=getPalette(length(unique(samples$Crucero)))) +
  facet_grid(~Crucero)

png(paste0(path,"/", "Figura-11-rarefaction2.png"), units="px", width=3700, height=2500, res=400)
plot(plot, col=adjustcolor("black", alpha=0.2))
dev.off()

```

```{r Figura 12. Div. alfa, fig.align = "center", fig.cap = "Figura 12. Diversidad alfa por estación. Se muestran los índices de diversidad alfa de Shannon e Inverso de Simpson calculados para la comunidad de zooplancton cada estación.", out.width='\\textwidth', fig.width = 7, fig.height = 5}

library(RColorBrewer)

getPalette = colorRampPalette(brewer.pal(3, "Paired"))
alfa <- phyloseq0
n <-length(unique(sample_data(alfa)$Corrida))
alfa <- transform_sample_counts(alfa, function(x) 1E6 * x/sum(x))

prichnes <- plot_richness(alfa, x="Estación", color="Crucero", measures=c("Shannon", "InvSimpson"))

prichnes$layers <- prichnes$layers[-1]

plot <- NULL
plot <- prichnes + scale_color_manual(values=getPalette(n)) + 
    geom_point(size=5, alpha=0.7) +
    geom_line(size=1, alpha=0.6, linetype=1, mapping = aes(group = Crucero)) +
        ylab("Diversidad alfa") +
        labs(title = paste0(set, " Cruceros")) + 
  #facet_grid(variable~Crucero, scales = 'free') + 
  #theme_bw() + 
  theme(axis.text.x = element_text(size = 12, angle = 90))
          
png(paste0(path,"/", "Figura-12-alfa-div.png"), units="px", width=4500, height=2500, res=400)
plot(plot, col=adjustcolor("black", alpha=0.2))
dev.off()



# richness <- estimate_richness(alfa, measures=c("Shannon", "InvSimpson"))

```


```{r Tabla 1. Índices Diversidad, eval = FALSE, echo = FALSE}
library(phyloseq)

richness <- estimate_richness(physeq, measures = c("Observed", 
                                   "Chao1", "ACE", 
                                   "Fisher", "Shannon", 
                                   "InvSimpson"
                                   ))

# rownames(richness) <- sample_data(physeq)$`Estación`

save <- round(richness[order(richness$Shannon, decreasing = TRUE),], digits = 2)

write.table(save, file = paste0(path, "/","AMB_richness.csv"))

# knitr::kable(save, caption = 'Tabla 1')

```


```{r Figura 13. PCoA, fig.align = "center", fig.cap = "Figura 13. PCoA. Componentes principales (Distancias Jaccard).", fig.width = 7, fig.height = 5}

getPalette = colorRampPalette(brewer.pal(3, "Paired"))
n <-length(unique(sample_data(phyloseq)$Crucero))

physeq.rr <- transform_sample_counts(phyloseq, function(x) 1E6 * x/sum(x))
iDist <- phyloseq::distance(physeq.rr, method="jaccard") # jaccard
iMDS  <- ordinate(rr, "NMDS", distance=iDist) 
# pairs(iMDS$values)
# hist(iMDS$values[, 'Eigenvalues'])
p <- plot_ordination(rr, iMDS, justDF = FALSE,
                     color="Crucero"
                     #label="Estación"
                     #shape="Cruice"
                     ) 
p$layers <- p$layers[-1]

plot <- NULL
plot <- p + geom_text(aes(label = Estación), size = 5, check_overlap = TRUE) + theme(legend.position = "top") +
  scale_color_manual(values=getPalette(n))
         

png(paste0(path,"/", "Figura-13-jaccard_NMDS.png"), units="px", width=3000, height=3000, res=400)
plot(plot, col=adjustcolor("black", alpha=0.2))
dev.off()

```


```{r deseq2}
library("DESeq2")


phy2deseq <- function (physeq, design, ...) 
{
    if (is.null(sample_data(physeq, FALSE))) {
        stop("There must be sample_data present, for specifying experimental design. See ?phyloseq_to_deseq2")
    }
    if (!taxa_are_rows(physeq)) {
        physeq <- t(physeq)
    }
    countData = round(as(otu_table(physeq), "matrix"), digits = 0)
    colData = data.frame(sample_data(physeq))
    if (requireNamespace("DESeq2")) {
        dds <- DESeq2::DESeqDataSetFromMatrix(t(countData), colData, 
            design, ...)
        return(dds)
    }
}


# define comparisons methods
levels(sample_data(rr)$Crucero)
contrast = contrasts(factor(sample_data(rr)$Crucero))
print(contrast)

X5_vs_X4 = subset_samples(rr, Crucero==c('X05','X04'))
X5_vs_X6 = subset_samples(rr, Crucero==c('X05','X06'))
X6_vs_X4 = subset_samples(rr, Crucero==c('X06','X04'))

X5_vs_X4 <- prune_taxa(taxa_sums(X5_vs_X4) > 1, X5_vs_X4)
X5_vs_X6 <- prune_taxa(taxa_sums(X5_vs_X6) > 1, X5_vs_X6)
X6_vs_X4 <- prune_taxa(taxa_sums(X6_vs_X4) > 1, X6_vs_X4)

y = c("X4_vs_X5")
data = X5_vs_X4

countData = round(as(otu_table(data), "matrix"), digits = 0)
colData = data.frame(sample_data(data))
dds <- DESeq2::DESeqDataSetFromMatrix(countData, colData, ~Crucero+Corrida)

deseq_obj = DESeq(dds, test="Wald", fitType="parametric")

res = results(deseq_obj, cooksCutoff = FALSE)
# alpha = 0.01
# sigtab = res[which(res$padj < alpha), ] # solve filter here!
sigtab = na.omit(res)
sigtab = cbind(as(sigtab, "data.frame"), as(tax_table(data)[rownames(sigtab), ], "matrix"))

sigtab['Cruice'] <-'NA'
sigtab[which(sigtab['log2FoldChange'] > 0 ),'Cruice'] <- "X04"
sigtab[which(sigtab['log2FoldChange'] < 0 ),'Cruice'] <- "X05"

# there's no good-significancy
# sigtab[which(sigtab['pvalue'] < 0.05 & abs(sigtab['log2FoldChange']) > 0), 'Signif'] <- 'PValue 0.05 | asb(logFC) <0'

write.table(sigtab, file=paste0(rank, "deseq_animalia.txt"), row.names =FALSE, sep="\t")
write.table(tax_table(Animalia_glom), file=paste0(rank, "animalia_glom2.txt"), row.names =FALSE, sep="\t")

# then plotplot 
theme_set(theme_bw())
scale_fill_discrete <- function(palname = "Set1", ...) {
    scale_fill_brewer(palette = palname, ...)
}

# Class order
x = tapply(sigtab$log2FoldChange, sigtab$Class, function(x) max(x))
x = sort(x, TRUE)
sigtab$Class = factor(as.character(sigtab$Class), levels=names(x))

# Family order
x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))
x = sort(x, TRUE)
sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))
# shape=Cruice
# svg(filename = paste0('deseq_in_',y, '.svg'),  width = 22, height = 14)
png(filename = paste0('deseq_in_',y, '.png'), height = 24,width = 60,res = 300,units = "cm")

d <- ggplot(sigtab, aes(x=Family, y=log2FoldChange, color=Class, shape=Cruice)) + geom_point(size=6) 
d <- d + theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5)) +
        ggtitle(paste0(" Plot of Differential Abundance ",rank, " in ", y," comparison"))
d + theme(text = element_text(size=18),panel.background = element_blank()) +
    geom_hline(yintercept=0, linetype="dashed", 
                color = "black", size=0.5)
dev.off()
```

## analyze only intersected samples

```{r select intersected samples}
samples$Estación <- as.character(samples$Estación)
intersect <- data.frame(logical = table(samples$Estación) == 3)
intersect$sample <- rownames(intersect)

getSamples_coi <- intersect[intersect$logical == TRUE,]$sample
samples0_coi <- samples[samples$Estación %in% getSamples_coi,]

seqtab.nochim0 <- seqtab.nochim[names(seqtab.nochim) %in% rownames(samples0_coi)]

phyloseq0 = phyloseq(otu_table(seqtab.nochim0, taxa_are_rows = TRUE), 
                    tax_table(as(tax, 'matrix')), 
                    sample_data(samples0_coi))

physeq0 <- prune_taxa(taxa_sums(phyloseq0) > 0, phyloseq0)

# Transform dataset to relative abundance
physeq2 <- NULL
# usar clean.obj o quitar zeros de physeq:
#physeq2 <- clean.obj
physeq2 <- transform_sample_counts(physeq0, function(x) (x / sum (x) ) )
# physeq2 <- phyloseq::filter_taxa(physeq2, function(x) var(x) > 1e-5, TRUE)
physeq2 <- transform_sample_counts(physeq2, function(x) x * 100 )
physeq2

```

```{r meta-diversity of 18S and COI}
physeq2

richness_coi <- estimate_richness(physeq2, measures=c("Shannon", "InvSimpson"))

DF_coi <- data.frame(richness_coi, sample_data(physeq2))

DF <- rbind(DF1, DF2, DF_coi)

mdf_multiplot = reshape2::melt(DF, measure.vars = c("Shannon", "InvSimpson"))


plot <- NULL
plot <- ggplot(mdf_multiplot, aes(y= value, x = Estación, color = Crucero)) + 
  scale_color_manual(values=getPalette(3)) + 
  #scale_color_manual(values='#B2DF8A') +
  geom_point(size=5, alpha=0.7) +
  geom_line(size=1, alpha=0.6, linetype=1, mapping = aes(group = Crucero)) +
  ylab("Diversidad alfa") +
  #labs(title = paste0(set, " Cruceros")) + 
  facet_wrap(marcador~variable, scales = 'free_y') + 
  theme_bw() + 
  theme(axis.text.x = element_text(size = 12, angle = 90))

png(paste0(path,"/", "Figura-12-alfa2-div-panel.png"), units="px", width=4000, height=2500, res=400)
plot(plot, col=adjustcolor("black", alpha=0.2))
dev.off()
```

```{r Figura 7S. intersected-samples-barplot.class, fig.align = "center", fig.cap = "Figura 7. Composición de las comunidades a nivel Clase a lo largo de las estaciones. Se presentan los taxones con varianza mayor a  1e-5.", out.width='\\textwidth', fig.width=8}


Class <-tax_glom(physeq2, taxrank="Clase")
Class = subset_taxa(Class, Clase!="Undetermined_R")

colourCount = length(unique(tax_table(Class)[,'Clase']))

library(RColorBrewer)

getPalette = colorRampPalette(brewer.pal(colourCount, "Paired"))

mdf = psmelt(Class)

p = ggplot(mdf, aes(x = Estación, y = Abundance, fill = Clase)) +
  scale_fill_manual(values = getPalette(colourCount)) + coord_flip()
p = p + geom_bar(stat = "identity", position = "stack", color = "black")
p = p + theme(axis.text.x = element_text(angle = -90, hjust = 0), 
              axis.text.y = element_text(size  = 10))

bar <- p + facet_grid(~ Crucero , space = "free", scales = "free") +
  labs(x = 'Estación', y = 'Abundancia Relativa (%)', 
         caption = "Abundancia relativa de taxones;  Se remueven las asignaciones 'Undetermined' por muestra") +
  guides(fill=guide_legend(ncol=1))



png(paste0(path,"/", "Figura-7-bar.Class_intersect_samples.png"), units="px", width=4200, height=3100, res=400)
plot(bar, col=adjustcolor("black", alpha=0.2))
dev.off()
```


```{r network}
ig <- make_network(physeq2, max.dist=0.4, type = 'taxa', dist.fun="bray")

plot_network(ig, phyloseq, color="Crucero", shape="Corrida", line_weight=0.4, label=NULL)
```

```{r ordination}
physeq2.ord <- ordinate(physeq2, "NMDS", "PCoA")
p1 = plot_ordination(rr, rr.ord, type="split", color="Filo", shape = 'Crucero')
p1 + facet_wrap(~Filo)
```


```{r Figura 13. PCoA, fig.align = "center", fig.cap = "Figura 13. PCoA. Componentes principales (Distancias Jaccard).", fig.width = 7, fig.height = 5}

getPalette = colorRampPalette(brewer.pal(3, "Paired"))
n <-length(unique(sample_data(physeq0)$Crucero))

physeq.rr <- transform_sample_counts(physeq0, function(x) 1E6 * x/sum(x))
iDist <- phyloseq::distance(physeq.rr, method="jaccard") # jaccard
iMDS  <- ordinate(physeq0, "PCoA", distance=iDist)

p <- plot_ordination(physeq0, iMDS, justDF = FALSE, color="Crucero") 

p$layers <- p$layers[-1]

plot <- NULL
plot <- p + geom_text(aes(label = Estación), size = 5, check_overlap = TRUE) + theme(legend.position = "top") +
  scale_color_manual(values=getPalette(n))
         

png(paste0(path,"/", "Figura-13-jaccard_pcoa_intersect.png"), units="px", width=3000, height=3000, res=400)
plot(plot, col=adjustcolor("black", alpha=0.2))
dev.off()

```

```{r superheat}
library(superheat)
Phylum <-tax_glom(Animalia.restore, taxrank="Filo")

heat.tbl <- otu_table(Phylum)
rownames(heat.tbl) <- tax_table(Phylum)[,2]
colnames(heat.tbl) <- sample_data(Phylum)$`Estación`
rowSums(heat.tbl)


# apply(heat.tbl, 1, function(x) (x  * sum(x) ))

superheat(heat.tbl, 
          # scale the variables/columns
          scale = FALSE,
          # change the color
          heat.pal = c( "#FFFFD9", "#081D58"),
          # change the color of the labels
          left.label.col = "white",
          # change the size of the label text
          left.label.text.size = 3,
          # change the angle of the label text
          bottom.label.text.angle = 90,
          bottom.label.text.size = 3,
          bottom.label.text.alignment = "right",
          bottom.label.col = "white",
          # add barplot next to the rows
          yr = log10(rowSums(heat.tbl)),
          yr.axis.name = "Numero de Secuencias\n(Log10)",
          yr.plot.type = "bar",
          # set bar colors
          yr.bar.col = "black",
          # Titles
          column.title = 'Muestras',
          column.title.size = 4,
          row.title = 'Taxones',
          row.title.size = 4,
          # remove the grid
          grid.hline = FALSE,
          grid.vline = FALSE
          )
```
